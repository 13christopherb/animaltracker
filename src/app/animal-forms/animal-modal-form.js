import React from 'react';
import {Field} from 'redux-form'
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {Button, Col, Form, Modal} from 'react-bootstrap';
import {Input, Select} from "../form-fields/index";

/**
 * Form for creating a new animal in the form of a modal
 * @param handleSubmit
 * @param handleChange
 * @param onSubmit
 * @param id
 * @param props
 * @returns {*}
 * @constructor
 */
export const AnimalModalForm = ({handleSubmit, handleChange, onSubmit, show, toggleModal, ...props}) => {
    return (
        <Modal show={show} onHide={toggleModal}>
            <Form onSubmit={handleSubmit(onSubmit)}>
                <Modal.Header closeButton>
                    <Modal.Title>New Animal</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <Form.Row>
                        <Col xs={4}>
                            {props.location ?
                                <Form.Group>
                                    <Form.Label>Location</Form.Label>
                                    <Form.Control as="select" defaultValue={props.location} disabled>
                                        <option>{props.location}</option>
                                    </Form.Control>
                                </Form.Group> :
                                <Field
                                    component={Select}
                                    id="location"
                                    title="Location"
                                    name="location"
                                    onChange={handleChange}
                                    options={[
                                        'NRO',
                                        'MBO',
                                        'SLO'
                                    ]}
                                />
                            }
                        </Col>
                    </Form.Row>
                    <Form.Row>
                        <Col xs={4}>
                            <Field
                                component={Select}
                                id="species"
                                title="Species"
                                name="species"
                                onChange={handleChange}
                                options={[
                                    'CSL',
                                    'ES',
                                    'HS',
                                    'NFS',
                                    'GFS',
                                    'SSO',
                                    'CET'
                                ]}
                            />
                        </Col>
                        <Col xs={8}>
                            <Field
                                component={Input}
                                id="name"
                                name="name"
                                title="Name"
                                type="text"
                                onChange={handleChange}
                            />
                        </Col>
                    </Form.Row>
                    <Form.Row>
                        <Col xs={5}>
                            <Field
                                component={Input}
                                id="weight"
                                name="weight"
                                title="Weight"
                                type="text"
                                onChange={handleChange}
                                append="kg"
                            />
                        </Col>
                    </Form.Row>
                    <Form.Row>
                        <Col xs={12}>
                            <Form.Group>
                                <Form.Label>
                                    Controlled meds:
                                    <Field
                                        component="input"
                                        name="isGettingControlledMeds"
                                        type="checkbox"
                                        onChange={handleChange}
                                    />
                                </Form.Label>
                            </Form.Group>
                            <Form.Label>
                                Tube feeding:
                                <Field
                                    component="input"
                                    name="isGettingTubed"
                                    type="checkbox"
                                    onChange={handleChange}
                                />
                            </Form.Label>
                        </Col>
                    </Form.Row>
                </Modal.Body>
                <Modal.Footer>
                    <Button variant="primary" onClick={toggleModal}
                            disabled={props.pristine || !props.valid}
                            type="submit">
                        Submit
                    </Button>
                    <Button variant="danger" onClick={toggleModal}
                            type="button">
                        Cancel
                    </Button>
                </Modal.Footer>
            </Form>
        </Modal>
    );
};