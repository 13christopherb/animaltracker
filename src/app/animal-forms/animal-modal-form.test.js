import React from 'react';
import configureMockStore from 'redux-mock-store';
import {mount, ReactWrapper} from 'enzyme';
import {AnimalModalForm} from './animal-modal-form';
import renderer from 'react-test-renderer'
import {Provider} from 'react-redux'
import {reduxForm} from 'redux-form'

const mockStore = configureMockStore();


jest.mock('react-dom');

const onSubmit = jest.fn();
const Decorated = reduxForm({
    form: 'animal-forms', onSubmit: {onSubmit}
})(AnimalModalForm);

describe('<AnimalModalForm />', () => {
    const defaultProps = {
        pristine: false,
        valid: true,
    };
    test('AnimalForm renders correctly', () => {
        const store = mockStore();
        const tree = renderer.create(
            <Provider store={store}>
                <Decorated
                    {...defaultProps}
                />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    });
    test('Disables button on form errors', () => {
        const store = mockStore();
        const props = {
            ...defaultProps,
            valid: false
        };
        const wrapper = mount(<Provider store={store}>
            <Decorated
                {...props}
            />
        </Provider>);
        const inside_els = document.getElementsByClassName("modal-body")[0];
        const inside_wrapper = new ReactWrapper(inside_els, true)
        expect(inside_wrapper.find('button').props().disabled).toEqual(true);
    });
});