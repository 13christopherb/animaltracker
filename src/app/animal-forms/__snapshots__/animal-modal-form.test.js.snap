// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimalModalForm /> AnimalForm renders correctly 1`] = `null`;
